framework/SupervisedLearning/ARMA.py

```diff
@@ -2020,15 +2021,19 @@ def getGlobalRomSegmentSettings(self, trainingDict, divisions):
           targetVals = trainingDict[target][0]
           periods = np.asarray(self.fourierParams[target])
           full = periods[periods > (delta*self.nyquistScalar)]
-          segment[target] = periods[np.logical_not(periods > (delta*self.nyquistScalar))]
-          if len(full):
-            # train Fourier on longer periods
-            self.fourierResults[target] = self._trainFourier(pivotValues, full, targetVals, target=target)
-            # remove longer signal from training data
-            signal = self.fourierResults[target]['predict']
-            targetVals = np.array(targetVals, dtype=np.float64)
-            targetVals -= signal
-            trainingDict[target][0] = targetVals
+          #NOTE: train all the Fourier periods in the segment.
+          segment[target] = periods
+          # # NOTE: orignially train the Fouirer whose periods shorter than segmentation length after segmentation. segment[target] store the shorter than segmentation length Fourier period.
+          #segment[target] = periods[np.logical_not(periods > delta)]
+          # if len(full):
+          #   # train Fourier on longer periods
+          #   self.fourierResults[target] = self._trainFourier(pivotValues, full, targetVals, target=target)
+          #   # self.fourierResults[target] = self._trainFourier(pivotValues, full, targetVals, masks=self._masks[self.zeroFilterTarget]['notZeroFilterMask'], target=target)
+          #   # remove longer signal from training data
+          #   signal = self.fourierResults[target]['predict']
+          #   targetVals = np.array(targetVals, dtype=np.float64)
+          #   targetVals -= signal
+          #   trainingDict[target][0] = targetVals
```

Comment:
it looks to me like this disables handling any Fourier fitting of the full signal before segmenting. Is that right?

Reply:
I believe some of these changes may have been necessary to get the fits you were using for the solar paper, that may have come from a branch of Andrea that never got merged to `devel`. If you want to close this and rebase, I would still maintain this PR and associated branch as a list of changes that "was working" in case you run into issues in the rebase.

Reply:
Create a new one is good to me. It seems this PR gets some weird behaviors.

Reply:
Ahh that makes sense Paul. I'll do that. Thanks!